### DISCUSSION TITLE
Throw the problematic exception node when a request exception occurs

### DISCUSSION THREAD

**Discussion by wang-qijia** [xetorthio/jedis#3466] [Ideas]
In a cluster or shard mode, it is often difficult to determine which node caused the exception when a request exception occurs, making it unfriendly for troubleshooting. It would be ideal to obtain the exception node for error logging or metric purpose.

Should we wrap the exception node in JedisConnectionException in the Connection class in order to record the error log more friendly?

**Comment by sazzad16** [comment]
@wang-qijia Thank you for sharing your idea. It's a interesting one.

**Comment by yangbodong22011** [comment]
@wang-qijia Good suggestion, I prepare a PR https://github.com/redis/jedis/pull/3470 to solve it.

**Reply by wang-qijia** [reply]
@yangbodong22011 Can we make the HostAndPort variables in the exception class so that the application layer can easily access them?

**Reply by yangbodong22011** [reply]
`JedisClusterOperationException` is used in scenarios where an exception occurs in the Cluster, but not all scenarios have a `legal` hostAndPort(e.g. processKey). I prefer to put it in the exception message when the hostAndPort exists.

**Reply by wang-qijia** [reply]
The Cluster API is just one of the options. There are other modes such as sharding and using the Jedis API. While the Cluster mode has some specific handling, it might be possible to handle other modes in the Connection class.

https://github.com/redis/jedis/blob/master/src/main/java/redis/clients/jedis/Connection.java#L165
